### PR TITLE
Total/Circulating-supply API

### PR DESCRIPTION
**Pull request by karthikeyan-cere**

### CODE REVIEW

File: api/index.js [hunk 1]
```diff
@@ -286,6 +287,28 @@ const limiter = rateLimit({
 
 app.use("/api/v1/faucet", limiter, faucet.faucet);
 
+app.get("/api/v1/cereTokens/:type", async(req, res) => {
```
Comment (AndreiNavoichyk) [inline api/index.js]: Let's use kebab convention

File: api/index.js [hunk 2]
```diff
@@ -286,6 +287,28 @@ const limiter = rateLimit({
...
+  try {
+    const { type } = req.params;
+    if (type === "totalSupply") {
+      const totalSupply = await ethNetwork.getCereTotalSupply();
+      return res.status(200).json(totalSupply);
+    } else if (type === 'circulating'){
```
Comment (AndreiNavoichyk) [inline api/index.js]: circulating-supply

File: api/index.js [hunk 3]
```diff
@@ -286,6 +287,28 @@ const limiter = rateLimit({
 
 app.use("/api/v1/faucet", limiter, faucet.faucet);
 
+app.get("/api/v1/cereTokens/:type", async(req, res) => {
+  try {
+    const { type } = req.params;
+    if (type === "totalSupply") {
```
Comment (AndreiNavoichyk) [inline api/index.js]: total-supply

File: api/index.js (hunk 1 above)
Comment (AndreiNavoichyk) [reply api/index.js]: Let's split into 2 routes

File: api/src/constants/config.js [hunk 4]
```diff
@@ -21,4 +21,5 @@ const config = {
 };
 
 const decimal = 10;
-module.exports = {config, decimal};
\ No newline at end of file
+const CereTokenContractAddress = "0x2da719db753dfa10a62e140f436e1d67f2ddb0d6";
```
Comment (AndreiNavoichyk) [inline api/src/constants/config.js]: Please, move to config

File: api/src/constants/cereErc20Abi.json [hunk 5]
```diff
@@ -0,0 +1,190 @@
+[
```
Comment (AndreiNavoichyk) [inline api/src/constants/cereErc20Abi.json]: Let's move to contracts

File: api/index.js (hunk 1 above)
Comment (karthikeyan-cere) [reply api/index.js]: Done

File: api/index.js (hunk 2 above)
Comment (karthikeyan-cere) [reply api/index.js]: Done

File: api/src/constants/cereErc20Abi.json (hunk 5 above)
Comment (karthikeyan-cere) [reply api/src/constants/cereErc20Abi.json]: Moved to contracts folder

File: api/src/constants/config.js (hunk 4 above)
Comment (karthikeyan-cere) [reply api/src/constants/config.js]: Moved to config

File: api/index.js (hunk 3 above)
Comment (karthikeyan-cere) [reply api/index.js]: Updated

File: api/index.js [hunk 6]
```diff
@@ -286,6 +287,9 @@ const limiter = rateLimit({
 
 app.use("/api/v1/faucet", limiter, faucet.faucet);
 
+app.get("/api/v1/cere-tokens/total-supply", ethNetwork.getCereTotalSupply);
```
Comment (AndreiNavoichyk) [inline api/index.js]: Rename to `/api/v1/cere-erc20-token/total-supply`
Please, use the same pattern for the second route

File: api/index.js [hunk 7]
```diff
@@ -286,6 +287,9 @@ const limiter = rateLimit({
...
+app.get("/api/v1/cere-tokens/circulating-supply", ethNetwork.getCereTotalCirculatingSupply);
```
Comment (AndreiNavoichyk) [inline api/index.js]: `getCereTotalCirculatingSupply` -> `getCereCirculatingSupply`

File: api/src/contracts/cereTokenAbi.json [hunk 8]
```diff
@@ -0,0 +1,190 @@
+[
```
Comment (AndreiNavoichyk) [inline api/src/contracts/cereTokenAbi.json]: Where did you take abi?

File: api/index.js (hunk 6 above)
Comment (karthikeyan-cere) [reply api/index.js]: Updated

File: api/index.js (hunk 7 above)
Comment (karthikeyan-cere) [reply api/index.js]: Updated

File: api/src/contracts/cereTokenAbi.json (hunk 8 above)
Comment (karthikeyan-cere) [reply api/src/contracts/cereTokenAbi.json]: [Etherescan](https://etherscan.io/address/0x2da719db753dfa10a62e140f436e1d67f2ddb0d6#code).

File: api/src/services/cereTokensService.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
+const { decimal } = require("../constants/config");
+const cereNetworkService = require('./cereNetworkService');
+const ethNetworkService = require('./ethNetworkService');
+
+const ethereumCERELockedAddresses = ["0x88E3386AC317892782f346e65FA7a4dd893552a1", "0x367ebF942F75E0c7Cf3EC0CEe1017349EDb3368b",
```
Comment (AndreiNavoichyk) [inline api/src/services/cereTokensService.js]: Load from `ETHEREUM_CERE_LOCKED_ADDRESSES`

**Review by AndreiNavoichyk — APPROVED** [review]

**Pull request merged**